ts/IterableInboxMessageDisplay.tsx

```diff
@@ -92,36 +92,29 @@ const IterableInboxMessageDisplay = ({
       let context = new IterableActionContext(action, source)
 
       Iterable.trackInAppClick(rowViewModel.inAppMessage, IterableInAppLocation.inbox, URL)
-
-      if (URL === 'iterable://delete') {
-         deleteRow(rowViewModel.inAppMessage.messageId)
-      }
-      
-      if(URL === 'iterable://dismiss') {
-         Iterable.trackInAppClose(rowViewModel.inAppMessage, IterableInAppLocation.inbox, IterableInAppCloseSource.link, URL)
-         returnToInbox()
-         return
-      }
-
-      if (URL.slice(0, 4) === 'http') {
-         Linking.openURL(URL)
-      }
-
       Iterable.trackInAppClose(rowViewModel.inAppMessage, IterableInAppLocation.inbox, IterableInAppCloseSource.link, URL) 
 
-      if(Iterable.savedConfig.urlHandler) {
-         Iterable.savedConfig.urlHandler(URL, context)
+      if (URL === 'iterable://delete') { 
+         returnToInbox(() => deleteRow(rowViewModel.inAppMessage.messageId))
+      } else if(URL === 'iterable://dismiss') {
+         returnToInbox(() => {})
+      } else if (URL.slice(0, 4) === 'http') {
+         returnToInbox(() => Linking.openURL(URL))
+      } else {
+         returnToInbox(() => {
+            if(Iterable.savedConfig.urlHandler) {
+               Iterable.savedConfig.urlHandler(URL, context)
+            } 
+         })
```

Comment:
I'm thinking this function (`handleHTMLMessage`) may need to be named differently now. It doesn't really connect to what it does, which is handling all actions from an in-app.